File: material/src/main/java/com/xadapter/material/AppBarStateChangeListener.java
package com.xadapter.material;

import com.google.android.material.appbar.AppBarLayout;

public class AppBarStateChangeListener implements AppBarLayout.OnOffsetChangedListener {

    public static final int EXPANDED = 0;
    public static final int COLLAPSED = 1;
    public static final int IDLE = 2;

    private int currentState = IDLE;

    @Override
    public void onOffsetChanged(AppBarLayout appBarLayout, int verticalOffset) {
        if (verticalOffset == 0) {
            currentState = EXPANDED;
        } else if (Math.abs(verticalOffset) >= appBarLayout.getTotalScrollRange()) {
            currentState = COLLAPSED;
        } else {
            currentState = IDLE;
        }
    }

    public int getCurrentState() {
        return currentState;
    }
}

File: material/src/main/java/com/xadapter/material/AppBarStateChangeListener.kt
package com.xadapter.material

import com.google.android.material.appbar.AppBarLayout
import kotlin.math.abs

/**
 * by y on 2016/11/17
 */
class AppBarStateChangeListener : AppBarLayout.OnOffsetChangedListener {

    var currentState = IDLE

    companion object {
        const val EXPANDED = 0
        const val COLLAPSED = 1
        const val IDLE = 2
    }

    override fun onOffsetChanged(appBarLayout: AppBarLayout, i: Int) {
        currentState = when {
            i == 0 -> EXPANDED
            abs(i) >= appBarLayout.totalScrollRange -> COLLAPSED
            else -> IDLE
        }
    }
}


